test/integration/server-test.js

```diff
@@ -20,13 +20,18 @@ test('server create event with reftype = tag', t => {
   const handleCreateEvent = robotMock.on.lastCall.args[1]
   t.is(typeof handleCreateEvent, 'function')
 
+  const configure = function (yaml) {
+    return {repository: 'name', labels: 'label'}
+  }
+
   handleCreateEvent({
     payload: {
       ref_type: 'tag',
       repository: {
         full_name: 'hoodiehq/first-timers-bot'
       }
-    }
+    },
+    config: configure
```

Comment:
you passed `context.config` in the "server create event with reftype = tag" test, but there are two more in this file that test `handleCreateEvent()`

You can set `--reporter=spec` to see more clearly which tests are passing/failing:

```
./node_modules/.bin/tap --reporter=spec test/integration/server-test.js
```

Reply:
The last test "server create event with non-existing branch name" is still failing for me, the reason is a racing condition. The assertion `t.is(githubMock.pendingMocks()[0], undefined)` is run before the request is made.

To workaround that, I’ve changed `getBranch(state)` in `handle-create-event.js` to `return getBranch(state)`. That allows us to do `await handleCreateEvent({...})` in the tests which will prevent the racing condition

Reply:
almost there! i'm getting that `await is a reserved word` again so that means i need to make a function async?